### PR TITLE
Add block storage config options

### PR DESCRIPTION
**Pull request by johnsca**
Depends on https://github.com/juju-solutions/interface-openstack-integration/pull/10
Addresses https://bugs.launchpad.net/charm-openstack-integrator/+bug/1809349

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -79,3 +79,34 @@ options:
...
+  ignore-volume-az:
+    description: |
+    type: boolean
+    default: null
+  bs-version:
+    description: |
+      Used to override automatic version detection. Valid values are v1, v2, v3
+      and auto. When auto is specified automatic detection will select the
+      highest supported version exposed by the underlying OpenStack cloud. If
+      not set, will use the upstream default.
```
Comment (hyperbolic2346) [inline config.yaml]: not sure I understand the upstream default vs auto. Why wouldn't I want the default to be auto?

**Comment by javacruft** [comment]
:+1: I think in principle what's being proposed looks OK

File: config.yaml (hunk 1 above)
Comment (johnsca) [reply config.yaml]: The default of `null` is intended to allow the upstream to provide the default by simply not rendering the config option.  This is to allow for the upstream default to potentially change, which doesn't much apply to this option, but it seemed easier to be consistent.

**Comment by tvansteenburgh** [comment]
@johnsca I just happened to be looking through old PRs, and am now wondering why `ignore-volume-az` is in `config.yaml` twice - I assume that's a mistake?

**Comment by johnsca** [comment]
@tvansteenburgh Yes, that was a mistake that got missed in review.  Linked PR addresses that.

**Pull request merged**